File: src/node/consumer/src/lib/repository/models/ingestedTorrent.ts
import {IIngestedTorrentAttributes, IIngestedTorrentCreationAttributes} from "@repository/interfaces/ingested_torrent_attributes";
import {Column, DataType, Model, Table} from 'sequelize-typescript';

const indexes = [
    {
        unique: true,
        name: 'ingested_torrent_unique_source_info_hash_constraint',
        fields: ['source', 'info_hash']
    }
];

@Table({modelName: 'ingested_torrent', timestamps: true, indexes: indexes})
export class IngestedTorrent extends Model<IIngestedTorrentAttributes, IIngestedTorrentCreationAttributes> {
    @Column({type: DataType.STRING(512)})
    declare name: string;

    @Column({type: DataType.STRING(512)})
    declare source: string;

    @Column({type: DataType.STRING(32)})
    declare category: string;

    @Column({type: DataType.STRING(64)})
    declare info_hash: string;

    @Column({type: DataType.STRING(32)})
    declare size: string;

    @Column({type: DataType.INTEGER})
    declare seeders: number;

    @Column({type: DataType.INTEGER})
    declare leechers: number;

    @Column({type: DataType.STRING(32)})
    declare imdb: string;

    @Column({type: DataType.BOOLEAN, defaultValue: false})
    declare processed: boolean;
}

File: src/shared/Models/IngestedTorrent.cs
namespace SharedContracts.Models;

public class IngestedTorrent
{
    public long? Id { get; set; }
    public string? Name { get; set; }
    public string? Source { get; set; }
    public string? Category { get; set; }
    public string? InfoHash { get; set; }
    public string? Size { get; set; }
    public int Seeders { get; set; }
    public int Leechers { get; set; }
    public string? Imdb { get; set; }

    public bool Processed { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    
    public string? RtnResponse { get; set; }
}
